main.go

```diff
@@ -0,0 +1,423 @@
+package main
+
+import (
+	"AntonKisselev/YP_GO_final/db"
+	"AntonKisselev/YP_GO_final/task"
+	"AntonKisselev/YP_GO_final/tests"
+	"encoding/json"
+	"github.com/joho/godotenv"
+	"io"
+	"log"
+	_ "modernc.org/sqlite"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+)
+
+func handlerApiNextDate(w http.ResponseWriter, req *http.Request) {
+	now := req.URL.Query().Get("now")
+	date := req.URL.Query().Get("date")
+	repeat := req.URL.Query().Get("repeat")
+
+	if now == "" || date == "" || repeat == "" {
+		w.WriteHeader(http.StatusBadRequest)
+	}
+
+	task := task.Task{
+		Repeat: repeat,
+		Date:   date,
+	}
+	nowTime, err := time.Parse("20060102", now)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte(err.Error()))
+		return
+	}
+	res, err := task.NextDate(nowTime)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte(err.Error()))
```

Comment:
Можно добавить немного информации для контекста
```suggestion
		w.Write([]byte("error in NextDate: " + err.Error()))
```